rq/connections.py

```diff
@@ -1,36 +1,479 @@
 # -*- coding: utf-8 -*-
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
-
+from redis.client import StrictRedis, Redis, StrictPipeline, Pipeline
 from contextlib import contextmanager
+import time
+import math
 
-from redis import StrictRedis
-
-from .compat.connections import patch_connection
-from .local import LocalStack, release_local
-
+from .local import LocalStack
+from .compat import string_types, as_text
+from .utils import import_attribute, compact, transaction
+from .keys import (QUEUES_KEY, queue_name_from_key, worker_key_from_name,
+                   WORKERS_KEY, queue_key_from_name, SUSPENDED_KEY)
+from .exceptions import NoSuchJobError
 
 class NoRedisConnectionException(Exception):
     pass
 
 
-@contextmanager
-def Connection(connection=None):
-    if connection is None:
-        connection = StrictRedis()
-    push_connection(connection)
-    try:
-        yield
-    finally:
-        popped = pop_connection()
-        assert popped == connection, \
-            'Unexpected Redis connection was popped off the stack. ' \
-            'Check your Redis connection setup.'
+class RQConnection(object):
+    from .worker import Worker
```

Comment:
This class statements executed during module import. So moving your import doesn't change the circle. You can define worker as a string in the class attribute

```python
class RQConnection:
    worker_class = 'rq.worker.Worker'
```

Then use `import_attribute` from `rq.utils` similar as job deserialization do.